### PR TITLE
boot: check parameter syntax before initializing the system.

### LINKED ISSUE #3112: [distributed] Check argument syntax before doing anything else
If an unrecognized argument syntax is presented we should print error, help and exit.
## Expected Behavior

For ex.

```
minio server server1:/export/disk1 server2:/export/disk2 server3:/export/disk3 server4:/export/disk4
```

We should error out for this.

## Current Behavior

Right now we start one minio process with all disks as local disks.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

`minio server <parameters>`

`<parameters>` syntax should be checked before doing anything else.
## Motivation and Context

Details here https://github.com/minio/minio/issues/3112
## How Has This Been Tested?

manual testing for minio-fs, singlenode-xl, distributed-xl
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
  fixes #3112

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -276,6 +259,99 @@ func isDistributedSetup(eps []*url.URL) (isDist bool) {
 	return isDist
 }
 
+// We just exit for invalid endpoints.
+func checkEndpointsSyntax(eps []*url.URL, disks []string) {
+	for i, u := range eps {
+		// Scheme is "" for FS and singlenode-XL
+		if u.Scheme != "" && u.Scheme != "http" && u.Scheme != "https" {
```
Comment (harshavardhana) [inline cmd/server-main.go]: This not correct on windows when you specify C:\disk windows keeps "C" as part of Scheme.  You can possibly validate this on windows in this manner.  You have to handle this just like how getPath() handles the right path from looking at the list of URLs on windows.

**Comment by harshavardhana** [comment]
The windows changes are good but you need to check why its failing again.  It is still one of those `fatalf`

**Comment by harshavardhana** [comment]
Can you add some test cases here @krishnasrinivas ?

**Comment by krishnasrinivas** [comment]
> Can you add some test cases here @krishnasrinivas ?

yes

**Comment by krishnasrinivas** [comment]
don't merge this yet, some parsing problem in windows

File: cmd/server-main.go [hunk 2]
```diff
@@ -119,8 +119,7 @@ func parseStorageEndpoints(eps []string) (endpoints []*url.URL, err error) {
 		if ep == "" {
 			return nil, errInvalidArgument
 		}
-		var u *url.URL
-		u, err = url.Parse(ep)
+		u, err := url.Parse(ep)
```
Comment (harshavardhana) [inline cmd/server-main.go]: This causes shadowing with "err error" return value. We need to keep it the same way as before or not return "err error" .

Comment (krishnasrinivas) [reply cmd/server-main.go]: ok

**Comment by krishnasrinivas** [comment]
> don't merge this yet, some parsing problem in windows

not related to this patch, will send separate fix later

**Pull request merged**